exec/cnex/global.c

```diff
@@ -481,19 +481,19 @@ Cell *global_getVariable(const char *pszVar)
     fatal_error("global_getVariable(): \"%s\" - invalid predefined variable.", pszVar);
 }
 
-void global_callFunction(const char *pszFunc, struct tagTExecutor *exec)
+void (*global_lookupFunction(const char *pszFunc, uint32_t *index))(struct tagTExecutor *exec)
 {
     uint32_t i;
 
     i = 0;
     while (gfuncDispatch[i].name) {
         if (strcmp(pszFunc, gfuncDispatch[i].name) == 0) {
-            (*gfuncDispatch[i].func)(exec);
-            return;
+            *index = i;
```

Comment:
I'm unclear here on why you need to return both the function pointer *and* its index. Wouldn't it be enough to just return one of them?

Reply:
Returning the index is just a diagnostic thing.  It tells me how many linear lookups we had to do to execute the code.  It's not _entirely_ necessary, but I'd like to keep it there, at least for a while.  Probably pull it out at some point.

Reply:
My only other option is to continue to pass a pointer to the `TExecutor` into this function, so I don't have to return the index.  Is that the better option?

Reply:
Fixed.